### PR TITLE
feat: migrate to Atomic V2

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/CDX-1281

## Proposed changes

Migrates the generated Atomic project to use Atomic v2

## Breaking changes

All the v2 breaking changes are included. **We should time the release of this feature shortly after the HSP (on https://github.com/coveo/search-interface-service) has done its Atomic V2 migration**. If users download an HSP using Atomic v2 **without upgrading the CLI**, they will have to migrate the local project themselves.

## Testing

- [X] Unit Tests:

- [X] Functionnal Tests:

- [X] Manual Tests:

[CDX-1281]: https://coveord.atlassian.net/browse/CDX-1281?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by olamothe — APPROVED** [review]

File: packages/cli-e2e/__tests__/atomic.specs.ts [hunk 1]
```diff
@@ -169,15 +169,6 @@ describe('ui:create:atomic', () => {
...
     },
-    {
-      describeName: 'when using an existing pageId (--pageId flag specified)',
-      buildAppOptions: {
-        id: 'with-page-id',
-        pageId: 'fffaafcc-6863-46cb-aca3-97522fcc0f5d',
-        skipInstall: true,
-      },
-      skipBrowser: true,
-    },
```
Comment (louis-bompart) [inline packages/cli-e2e/__tests__/atomic.specs.ts:180]: So we stop testing that? :suspect:

Comment (ThibodeauJF) [reply packages/cli-e2e/__tests__/atomic.specs.ts:180]: When I looked at it last year it seemed like a duplicate. Now I'm not so sure...

Comment (louis-bompart) [reply packages/cli-e2e/__tests__/atomic.specs.ts:180]: After DM, yep you're right it's a dupe.
LGTM!

**Review by louis-bompart — APPROVED** [review]

**Review by btaillon — APPROVED** [review]

File: packages/ui/atomic/template/README.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ The root folder should contain a `.env` file. Replace all placeholder variables
 
 ### CDN
 
-By default, the project installs the latest major Atomic version, v1, to allow types and more advanced customizations. [Coveo Headless](https://www.npmjs.com/package/@coveo/headless) is also bundled with Atomic and accessible at `@coveo/atomic/headless`.
+By default, the project installs the latest major Atomic version, v2, to allow types and more advanced customizations. [Coveo Headless](https://www.npmjs.com/package/@coveo/headless) is also bundled with Atomic and accessible at `@coveo/headless`.
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/ui/atomic/template/README.md:11]: good point. I would just change bundled for installed and GG
```suggestion
By default, the project installs the latest major Atomic version, v2, to allow types and more advanced customizations. [Coveo Headless](https://www.npmjs.com/package/@coveo/headless) is also installed with Atomic and accessible at `@coveo/headless`.
```

File: packages/ui/atomic/template/README.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ The root folder should contain a `.env` file. Replace all placeholder variables
...
 
 When running, the app will use the bundled Atomic, but using the CDN is a viable option, just make sure you're using the same minor version of Atomic as the one bundled. It could cause issues with your custom components if the minor version differs.
 E.g., if you have @coveo/atomic@1.44.0 installed, use the following CDN link at path [https://static.cloud.coveo.com/atomic/v1.44/](https://static.cloud.coveo.com/atomic/v1.44/atomic.esm.js).
```
Comment (louis-bompart) [inline packages/ui/atomic/template/README.md:14]:
```suggestion
E.g., if you have @coveo/atomic@2.0.0 installed, use the following CDN link at path [https://static.cloud.coveo.com/atomic/v2.0/](https://static.cloud.coveo.com/atomic/v2.0/atomic.esm.js).
```

Just so we don't have v1 mention anymore

**Pull request merged**